disperser/dataapi/metrics.go

```diff
@@ -101,35 +104,124 @@ type DynamoDBCollector struct {
 	blobMetadataStore *blobstore.BlobMetadataStore
 	blobStatusMetric  *prometheus.Desc
 	logger            common.Logger
+	metricsCache      map[disperser.BlobStatus]int
+	cacheMutex        sync.RWMutex
 }
 
 func NewDynamoDBCollector(blobMetadataStore *blobstore.BlobMetadataStore, logger common.Logger) *DynamoDBCollector {
-	return &DynamoDBCollector{
+	collector := &DynamoDBCollector{
 		blobMetadataStore: blobMetadataStore,
 		blobStatusMetric: prometheus.NewDesc("dynamodb_blob_metadata_status_count",
 			"Number of blobs with specific status in DynamoDB",
 			[]string{"status"},
 			nil,
 		),
-		logger: logger,
+		logger:       logger,
+		metricsCache: make(map[disperser.BlobStatus]int),
 	}
+
+	if blobMetadataStore != nil {
+		go collector.periodicFetch()
```

Comment:
instead of creating a new goroutine from the constructor, it's generally preferable to return the instance and have a separate, explicit `Start` method that starts the goroutine or runs the infinite loop with a context that the caller can call in a new goroutine